### PR TITLE
Update for golang1.17

### PR DESCRIPTION
**Pull request by elmiko**
This change updates the go.mod, makefile, dockerfiles, and the vendor directory to use golang 1.17 for building. it also adds a minor convenience option to the makefile that allows a user to override the build image.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,3 +32,83 @@ require (
 	sigs.k8s.io/controller-tools v0.6.3-0.20210916130746-94401651a6c3
 	sigs.k8s.io/yaml v1.2.0
 )
+
+require (
```
Comment (JoelSpeed) [inline go.mod:36]: Is this a new thing for go modules in 1.17?

Comment (elmiko) [reply go.mod:36]: seems like it, i ran `make vendor` and it produces this

**Comment by JoelSpeed** [comment]
This needs an update to the `tools.go` build tags to pass the go1.17 build vetting

**Comment by elmiko** [comment]
ack, thanks

**Comment by elmiko** [comment]
update
* ran `make check` to reformat any code that needed it

**Comment by elmiko** [comment]
update
* update a missed image in Dockerfile.rhel

**Pull request merged**